File: openondemand/rstudio/examples/01_hello_cluster.R
# Hello Cluster - First R Script on KENET HPC
# This script demonstrates basic R commands and cluster information

cat("=" %R% rep("=", 60) %R% "\n")
cat("Hello from KENET HPC Cluster!\n")
cat("=" %R% rep("=", 60) %R% "\n\n")

# System Information
cat("System Information:\n")
cat("  R Version:", R.version.string, "\n")
cat("  Platform:", R.version$platform, "\n")
cat("  Hostname:", Sys.info()["nodename"], "\n")
cat("  OS:", Sys.info()["sysname"], Sys.info()["release"], "\n")

# User Information
cat("\nUser Information:\n")
cat("  Username:", Sys.info()["user"], "\n")
cat("  Home Directory:", Sys.getenv("HOME"), "\n")
cat("  Working Directory:", getwd(), "\n")

# Date and Time
cat("\nCurrent Time:", format(Sys.time(), "%Y-%m-%d %H:%M:%S"), "\n")

# Available CPU Cores
cat("\nComputational Resources:\n")
cat("  Available CPU Cores:", parallel::detectCores(), "\n")

# Memory Information
if (require(pryr, quietly = TRUE)) {
  cat("  Available Memory:", pryr::mem_used() / 1e9, "GB used\n")
} else {
  cat("  Install 'pryr' package for memory information\n")
}

# Check for common packages
cat("\nChecking installed packages:\n")
packages_to_check <- c("tidyverse", "ggplot2", "dplyr", "readr", 
                       "data.table", "rmarkdown", "knitr")

for (pkg in packages_to_check) {
  if (requireNamespace(pkg, quietly = TRUE)) {
    version <- as.character(packageVersion(pkg))
    cat("  ✓", pkg, version, "\n")
  } else {
    cat("  ✗", pkg, "not installed\n")
  }
}

# Simple calculation
cat("\nSimple Calculation:\n")
numbers <- 1:100
cat("  Sum of 1 to 100:", sum(numbers), "\n")
cat("  Mean:", mean(numbers), "\n")
cat("  Standard Deviation:", sd(numbers), "\n")

cat("\n" %R% rep("=", 60) %R% "\n")
cat("Script completed successfully!\n")
cat("=" %R% rep("=", 60) %R% "\n")

File: openondemand/vscode/examples/01_hello_cluster.py
#!/usr/bin/env python3
"""
Simple Python script to test VS Code on KENET HPC cluster
This demonstrates basic Python execution and cluster information
"""

import os
import sys
import socket
import platform
from datetime import datetime

def main():
    print("=" * 60)
    print("Hello from KENET HPC Cluster!")
    print("=" * 60)
    
    # System information
    print(f"\nSystem Information:")
    print(f"  Hostname: {socket.gethostname()}")
    print(f"  Platform: {platform.platform()}")
    print(f"  Python Version: {sys.version}")
    print(f"  Python Executable: {sys.executable}")
    
    # User and environment
    print(f"\nUser Information:")
    print(f"  Username: {os.getenv('USER')}")
    print(f"  Home Directory: {os.getenv('HOME')}")
    print(f"  Current Directory: {os.getcwd()}")
    
    # Date and time
    print(f"\nCurrent Time: {datetime.now().strftime('%Y-%m-%d %H:%M:%S')}")
    
    # Check for common libraries
    print(f"\nChecking installed packages:")
    try:
        import numpy
        print(f"   NumPy {numpy.__version__}")
    except ImportError:
        print(f"   NumPy not installed")
    
    try:
        import pandas
        print(f"  ✓ Pandas {pandas.__version__}")
    except ImportError:
        print(f"   Pandas not installed")
    
    try:
        import matplotlib
        print(f"   Matplotlib {matplotlib.__version__}")
    except ImportError:
        print(f"   Matplotlib not installed")
    
    print("\n" + "=" * 60)
    print("Script completed successfully!")
    print("=" * 60)

if __name__ == "__main__":
    main()
